Fix npm vulnerabilities

tomalec opened the pull request:
<!-- You can erase any parts of this template not applicable to your pull request. -->


Our packages deliver `132 vulnerabilities (2 low, 76 moderate, 50 high, 4 critical)`
and a number of packages are pretty outdated
![image](https://user-images.githubusercontent.com/17435/139253603-dcd31b33-57ad-49a8-ad5b-860e5a11aca6.png)


This PR aims to address at least a part of it.

 Our lowest WordPress 5.6 theoretically supports https://github.com/WordPress/wordpress-develop/blob/5.6/package.json#L80 so we cannot bump our `@wordpress/` dependencies.
WooCommerce 5.3.2, uses WC-admin `2.2.7`. Now we need some excavation to find out the individual packages. But it seems we use already higher ones. Maybe we could consider bumping the lowest WC to L-2 = 5.6 now?

So we only can bump devDependencies (and probably not all)

I'm not a user of gulp flows, so I skipped it as well.
Personally, I'd replace `underscore` with native modern ES, but that's something for a bigger PR.

Bumping es/css lints would require adapting to new rules, so it's another candidate for a separate PR.


### Changes proposed in this pull request
<!-- Explain what this PR adds or changes, why, and how this will impact users. -->

- Run `npm audit fix` (562d5a0)
- Add `repository` field to `package.json`, (1605d76)
to suppress warning during `npm install`.

#### Screenshots
##### Before
`npm install`
![132 vulnerabilities (2 low, 76 moderate, 50 high, 4 critical)](https://user-images.githubusercontent.com/17435/139056652-e548e466-4a7b-4a22-80b8-faa59dc04f1a.png)
`npm outdated`
![image](https://user-images.githubusercontent.com/17435/139056387-a938051e-1535-49d4-914d-81230d17371b.png)
##### After
![found 60 vulnerabilities (1 low, 11 moderate, 44 high, 4 critical)
](https://user-images.githubusercontent.com/17435/139253246-aa15a982-4437-4630-9354-733d4e49ebb0.png)


### Detailed test instructions
<!-- Add steps to confirm the fix or change. -->
1. `npm install`

<!-- Please add details of other areas that might be impacted by the change. -->

### Changelog note

tomalec committed:
Run `npm audit fix`

tomalec committed:
Add `repository` field to `package.json`,

to suppress warning during `npm install`.

eason9487 commented on package.json:
```diff
@@ -4,6 +4,10 @@
+	"repository": {
+		"type": "git",
+		"url": "git@github.com:woocommerce/pinterest-for-woocommerce.git"
+	},
```
💅 
Mixed tab and space indentations.

eason9487 reviewed (approved):
LGTM and approving in advance. Left a comment for a minor issue that has mixed indentations.

tomalec committed:
Fix whitespace consistency in `package.json`

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/233#discussion_r739134989
